src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -201,20 +214,24 @@ struct RemoveUnusedModuleElements : public Pass {
       return analyzer.reachable.count(
                ModuleElement(ModuleElementKind::Event, curr->name)) == 0;
     });
-
-    for (auto& table : module->tables) {
-      table->segments.erase(
-        std::remove_if(table->segments.begin(),
-                       table->segments.end(),
-                       [&](auto& seg) { return seg.data.empty(); }),
-        table->segments.end());
-    }
+    module->removeElementSegments([&](ElementSegment* curr) {
+      return curr->data.empty() ||
+             analyzer.reachable.count(ModuleElement(
+               ModuleElementKind::ElementSegment, curr->name)) == 0;
+    });
+    std::unordered_set<Name> nonemptyTables;
+    ModuleUtils::iterActiveElementSegments(*module, [&](ElementSegment* elem) {
+      if (nonemptyTables.count(elem->table) == 0) {
+        nonemptyTables.insert(elem->table);
```

Comment:
I added a comment above to make it more clear, but since we are removing empty element segments above this, all tables that have remaining segments need to be kept. Removed the redundant `if`.